### DISCUSSION TITLE
How can we use the library in our own code with import, instead of cloning the whole project?

### DISCUSSION THREAD

**Discussion by sming256** [open-mmlab/mmaction2#2508] [Q&A]
Thanks for your amazing repo and great effort in the video understanding field. I have a question about importing the existing model.

Suppose I want to create a new model with minor changes to an existing model, and use it in other tasks. So I would prefer to install `mmaction2`, and call it as a library, such as the following.
```python
import mmaction
import torch.nn as nn
from mmaction.registry import MODELS

@MODELS.register_module()
class mymodel(nn.Module):
    def __init_(self):
        pass

    def forward(self):
        pass

print(MODELS)
```
However, when I print the MODELS, I only have my new model. 
```
           Registry of model            
┏━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Names   ┃ Objects                    ┃
┡━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ mymodel │ <class '__main__.mymodel'> │
└─────────┴────────────────────────────┘
```
 How can I import the existing model and also keep my customized model?

**Comment by cir7** [comment]
you can call `register_all_modules` defined [here](https://github.com/open-mmlab/mmaction2/blob/main/mmaction/utils/setup_env.py#L8) to register MMAction2 modules at any position you want to use them
